.ci/release/Jenkinsfile

```
@@ -0,0 +1,229 @@
+                  nexusCloseStagingRepository(stagingProfileId: "${SPID}", stagingId: foundStagingId)
+                  nexusReleaseStagingRepository(stagingProfileId: "${SPID}", stagingId: foundStagingId)
+                }
+              }
+            }
+          }
+        }
+        stage('Branch creation') {
+          steps {
+            dir("${BASE_DIR}") {
+              setupAPMGitEmail()
+              script {
+                env.BRANCH_NAME = "tags/${env.TAG_BARE}"
+              }
+              withGitRelease() {
+                script {
+                  def curVer = mvnVersion()
+                  sh(script: "./scripts/jenkins/branch_creation.sh ${curVer}")
+                  def isMajor = sh(script: "./scripts/jenkins/is_major.sh ${curVer}", returnStatus: true)
+                  if (isMajor == 0) {
+                    input message: "This was a major version release. Please update the conf.yml in the docs repo before continuing", ok "Continue"
+                  }
+                }
+                gitPush()
+              } 
+            }
+          }
+        }
+        stage('Create GitHub release draft') {
+          steps {
+            deleteDir()
+            unstash 'source'
+            dir("${BASE_DIR}"){
+              script {
+                def curVer = mvnVersion(showQualifiers: false)
+                // Construct the URL with anchor for the release notes
+                // Ex: https://www.elastic.co/guide/en/apm/agent/java/current/release-notes-1.x.html#release-notes-1.13.0
+                def finalUrl = sh(script: "scripts/jenkins/generate_release_notes_url.sh ${curVer}", , returnStdout: true)
+                def ret = githubReleaseCreate(draft: true, name: "Release ${curVer}", body: "[Release Notes for ${curVer}](${finalUrl})")
+                env.RELEASE_ID = ret['id']
+                env.RELEASE_NOTES_URL = finalUrl
+              }
+            }
+
+          }
+        }
+        stage('Wait for artifact to be available in Maven Central') {
+          steps {
+            script {
+              def fullUrl = sh(script: "./scripts/jenkins/maven_artifact_url.sh ")
+              echo "Sleeping for ten minutes to wait for Sonatype"
+              sleep(time: 10, unit: "MINUTES")
+              waitUntil(initialRecurrencePeriod: 60000) {
+                script {
+                  def ret = sh(script: "curl -fs $fullUrl >/dev/null 2>&1", returnStatus: true)
+                  echo "Waiting for the artifacts to be published on Sonatype"
+                  return ret == 0
+                }
+              }
+            }
+          }
+        }
+        stage('Update Cloudfoundry') {
+          steps {
+            deleteDir()
+            unstash 'source'
+            dir("${BASE_DIR}"){
+              script {
+                env.BRANCH_NAME = "master"
+              }
+              withGitRelease() {
+                setupAPMGitEmail()
```

See my previous comments
```suggestion
              withGitRelease() {
```